### PR TITLE
ci: When models change, dump gql schema

### PR DESCRIPTION
**Pull request by Yaminyam**
To easily identify when the graphql schema has changed, a dump is created when files in the models path are modified.

### CODE REVIEW

**Review by achimnol — CHANGES_REQUESTED** [review]
Let's generalize the workflow filename and the top-level `name` key to `update-api-schema`.
Let's rename the job to `graphql`.

This will give a concise yet extensible check name "update-api-schema / graphql" where we could add "update-api-schema / rest" in the future.

**Comment by achimnol** [comment]
As of #1686, we should use `mgr api dump-gql-schema` instead of `mgr gql show`.

File: .github/workflows/update-api-schema.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      if: ${{ env.REMOTE_CACHE_BACKEND_ENDPOINT != '' }}
+    - name: Bootstrap Pants
+      uses: ./actions/init-pants
+      # See: github.com/pantsbuild/actions/tree/main/init-pants/
+      # ref) https://github.com/pantsbuild/example-python/blob/main/.github/workflows/pants.yaml#L30-L49
+      with:
+        named-caches-hash: ${{ hashFiles('python*.lock', 'tools/*.lock') }}
+        cache-lmdb-store: 'true'
+    - name: Pants export
+      run: pants export '::'
```
Comment (achimnol) [inline .github/workflows/update-api-schema.yml]: We don't need to export all venvs to run the `mgr` commands.
Let's change it to `pants export --resolve=python-default`.

File: .github/workflows/update-api-schema.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - name: Create GraphQL schema dump
+      run: |
+        ./backend.ai mgr gql show >> schema.graphql
```
Comment (achimnol) [inline .github/workflows/update-api-schema.yml]: Please set the correct destination path as suggested in the discussion thread.

**Review by achimnol — CHANGES_REQUESTED** [review]

File: .github/workflows/update-api-schema.yml (hunk 2 above)
Comment (Yaminyam) [reply .github/workflows/update-api-schema.yml]: This content will be reflected by modifying it to use the --output option after number 1695 is merged.

**Comment by Yaminyam** [comment]
> * [x]  Let's also migrate the OpenAPI XML generator (`./py -m ai.backend.manager.openapi`) to `mgr openapi show` command.
As with graphql, it was decided to replace this with mgr api dump-openapi.

**Comment by achimnol** [comment]
Please resolve the merge conflict. I think the conflict is due to the formatting rule change in #1475, so you may `git checkout --ours`, check if there are any missing attributes added in the main branch, and reformat it again using `pants fmt`.

**Pull request merged**